### ISSUE TITLE
minimumCoverageThresholds:lineCoverage not working

### ISSUE THREAD

**Issue by markusroessler** [danielpalme/ReportGenerator#653]
**Describe the bug**
When executing reportgenerator with `--minimumCoverageThresholds:lineCoverage=100`on a cobertura report with `line-rate="0.1433"`, the exit code is 0 a no error is reported.
`--minimumCoverageThresholds:branchCoverage=100` and `--minimumCoverageThresholds:methodCoverage=100` do report the coverage errors as expected
[Settings-Description](https://github.com/danielpalme/ReportGenerator/wiki/Settings)

**To Reproduce**
- Create a coverage.cobertura.xml report with line-rate below 1.0
- Execute `reportgenerator -reports:"**/TestResults/*/coverage.cobertura.xml" -reporttypes:Html --minimumCoverageThresholds:lineCoverage=100`

**Reportgenerator Version**
5.2.1.0

**Comment by danielpalme** [comment]
Will have a look within the next days.

**Comment by danielpalme** [comment]
I can't reproduce your problem.

What line coverage does your HTML report contain?
I referring to this card in the report:
![image](https://github.com/danielpalme/ReportGenerator/assets/859606/5deb6b3e-df4b-4995-9d3f-cb7178d1c02a)

Does this indicate a coverage of 100%?

The `line-rate` attribute in the Cobertura file is not relevant to evaluate if the coverage is below 100%. 
_ReportGenerator_ calculates its own line/branch/method coverage metrics. Those are relevant for the evaluation and are also visible in the HTML report.

If you like, you can also send me your Cobertura file via email: reportgenerator@palmmedia.de

**Comment by markusroessler** [comment]
here is an example cobertura file generated from a .NET project: [coverage.cobertura.xml.txt](https://github.com/danielpalme/ReportGenerator/files/14379875/coverage.cobertura.xml.txt)
The html report shows a line coverage of 50% - but reportgenerator completes with exit code 0.

**Comment by danielpalme** [comment]
I still can't reproduce the problem. I' m using this version: https://www.nuget.org/packages/dotnet-reportgenerator-globaltool/5.2.1

Executed command: `reportgenerator -reports:coverage.cobertura.xml -targetdir:output -reporttypes:Html --minimumCoverageThresholds:lineCoverage=100`

Output:
```bat
PS C:\temp> reportgenerator -reports:coverage.cobertura.xml -targetdir:output -reporttypes:Html --minimumCoverageThresholds:lineCoverage=100
2024-02-23T20:43:29: Arguments
2024-02-23T20:43:29:  -reports:coverage.cobertura.xml
2024-02-23T20:43:29:  -targetdir:output
2024-02-23T20:43:29:  -reporttypes:Html
2024-02-23T20:43:29:  --minimumCoverageThresholds:lineCoverage=100
2024-02-23T20:43:29: File '\Users\mroessler\Workspaces\Samples\ReportgeneratorLineCoverageSample\Sample\Class1.cs' does not exist (any more).
2024-02-23T20:43:29: Writing report file 'output\index.html'
2024-02-23T20:43:29: The line coverage of 50% is below the minimum threshold of 100%.
```

![image](https://github.com/danielpalme/ReportGenerator/assets/859606/cf1524c7-e7ac-490e-b6a0-8c5af251e171)

**Comment by markusroessler** [comment]
I just realised it actually works with Windows but not Mac OS (Version 5.2.1 an both platforms). 
Sorry for leaving out this information - I didn't expected this to be relevant.

**Comment by markusroessler** [comment]
Is it a language thing maybe? My Mac's system language is english currently - Windows is german

**Comment by danielpalme** [comment]
> Is it a language thing maybe? My Mac's system language is english currently - Windows is german

I don't think so. And the there is no difference in the implementation between lineCoverage, branchCoverage and methodCoverage

Will try to debug on Linux (I don't have a Mac available).

**Comment by markusroessler** [comment]
might have something to do with commandline parsing. This does NOT work on Mac:
`reportgenerator -reports:coverage.cobertura.xml -targetdir:output -reporttypes:Html --minimumCoverageThresholds:lineCoverage=100`           

This does work :) 
`reportgenerator -reports:coverage.cobertura.xml -targetdir:output -reporttypes:Html XXXXX --minimumCoverageThresholds:lineCoverage=100`   

Same happens for minimumCoverageThresholds:methodCoverage and branchCoverage - so not lineCoverage related. It just looks like the first "--"-argument is ignored.

**Comment by danielpalme** [comment]
Please try again with version 5.2.2. This new release should fix the issue.

**Comment by markusroessler** [comment]
I can confirm the issue is fixed in 5.2.2 on Mac OS. Thank you for the fast support!
